app/services/puppetdb.rb

```diff
@@ -1,15 +1,26 @@
 module Puppetdb
+
+  class UnsupportedVersion < StandardError; end
+
   def self.client
     options = {
       :uri => uri,
       :ssl_ca_file => Setting[:puppetdb_ssl_ca_file],
       :ssl_certificate_file => Setting[:puppetdb_ssl_certificate],
       :ssl_private_key_file => Setting[:puppetdb_ssl_private_key]
     }
-    if uri.path.start_with?('/pdb')
+
+    api_version = Setting[:puppetdb_api_version].to_i
+
+    case api_version
+    when 1
+      PuppetdbClient::V1.new(options)
+    when 3
       PuppetdbClient::V3.new(options)
+    when 4
+      PuppetdbClient::V4.new(options)
     else
-      PuppetdbClient::V1.new(options)
+      raise ::Puppetdb::UnsupportedVersion
```

Comment:
Please use `raise Foreman::Exception.new(N_('Unsupported PuppetDB version.'))` instead. This will generate an [ERF-Code](http://projects.theforeman.org/projects/foreman/wiki/ErrorCodes) for easier troubleshooting.